### PR TITLE
anti-bot

### PR DESCRIPTION
**Pull request by hkubus**

### CODE REVIEW

**Review by hkubus — COMMENTED** [review]
cool

File: src/cmds/config.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export async function run(
             .setDescription(
                 'Nie posiadasz permisji do wykonania tego!\nWymagane permisje: Manage Server'
             )
-            .setFooter(client.footer);
+            .setFooter({ text: client.footer });
```
Comment (SuperchupuDev) [inline src/cmds/config.ts:18]: didnt you plan to remove it outside the class?

File: src/events/messageCreate.ts [hunk 2]
```diff
@@ -13,37 +15,67 @@ export async function run(client: FluorineClient, message: Message) {
             )
         );
     }
+
     // @ts-ignore
     const settings: SettingsType = await r
         .table('config')
         .get(message.guild?.id)
         .run(client.conn);
+    if (settings.antibot !== 0) {
```
Comment (SuperchupuDev) [inline src/events/messageCreate.ts]:
```ts
if (settings.antibot) {
```
would do the same, use that

File: src/util/messageBot.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import FluorineClient from '@classes/Client';
+import { Message } from 'discord.js';
+
+export async function messageBot(client: FluorineClient, message: Message) {
+    let bot = 0;
+    const authorDate = new Date(message.author.createdTimestamp + 12096e5);
+    const memberDate = new Date(message.member.joinedTimestamp + 1800000);
+    const url = message.content.match(/\bhttps?:\/\/\S+/giu);
+    const urlBoolean = false;
```
Comment (SuperchupuDev) [inline src/util/messageBot.ts]: why is this a thing? it's only used on an if check which isn't ever triggered because it's always false

**Review by SuperchupuDev — DISMISSED** [review]
Some small things i noticed

File: src/api/routes/set.ts [hunk 4]
```diff
@@ -16,6 +16,9 @@ export default async function set(
         if (!user) {
             return res.status(401).send('Unauthorized');
         }
+        if (isNaN(req.body.antibot)) {
+            return res.send('Invalid antibot value');
```
Comment (SuperchupuDev) [inline src/api/routes/set.ts]:
```suggestion
            return res.status(400).send('Invalid antibot value');
```

File: src/classes/handlers/PhishingHandler.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                })
+            }
+        );
+        return request.json();
+    }
+    getWords() {
+        return this.word.split('\n');
+    }
+    getUsers() {
+        return this.word.split('\n');
```
Comment (SuperchupuDev) [inline src/classes/handlers/PhishingHandler.ts]:
```suggestion
        return this.users.split('\n');
```

File: src/util/messageBot.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    const currentDate = new Date();
+    const urls = [];
+    if (client.phishingUsers.includes(message.author.id)) {
+        bot += 15;
+    }
+    client.words.forEach(word => {
+        if (message.content.includes(word)) {
+            bot += 5;
+        }
+    });
```
Comment (SuperchupuDev) [inline src/util/messageBot.ts]:
```suggestion
    client.words
        .filter(word => message.content.includes(word))
        .forEach(() => bot += 5);
```

File: src/classes/handlers/PhishingHandler.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+import { fetch } from 'undici';
+export default class PhishingHandler {
+    word: string;
+    client: FluorineClient;
+    users: string;
+    constructor(client: FluorineClient) {
+        this.client = client;
+        this.users = readFileSync(
+            `${__dirname}/../../../assets/users.txt`
+        ).toString();
```
Comment (SuperchupuDev) [inline src/classes/handlers/PhishingHandler.ts:17]:
```suggestion
        );
```

File: src/classes/handlers/PhishingHandler.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        this.word = readFileSync(
+            `${__dirname}/../../../assets/words.txt`
+        ).toString();
```
Comment (SuperchupuDev) [inline src/classes/handlers/PhishingHandler.ts:20]:
```suggestion
        );
```

Comment (hkubus) [reply src/classes/handlers/PhishingHandler.ts:20]: if you dont put in .toString() im pretty sure ts returns an error

File: src/classes/handlers/PhishingHandler.ts (hunk 7 above)
Comment (hkubus) [reply src/classes/handlers/PhishingHandler.ts:17]: same with here

**Review by SuperchupuDev — DISMISSED** [review]

File: src/util/messageBot.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    const currentDate = new Date();
+
+    const urls = url?.map(link => {
+        bot += 5;
+        link = link.replaceAll('www.', '');
+        return { url: link };
+    });
+    const urlResponse = await client.phishing.getLink(urls);
+
+    if (urlResponse === {}) {
```
Comment (SuperchupuDev) [inline src/util/messageBot.ts]: this will always return false, otherwise LGTM

**Review by SuperchupuDev — CHANGES_REQUESTED** [review]

**Review by SuperchupuDev — APPROVED** [review]

**Pull request merged**